packages/astro/src/core/build/index.ts

```diff
@@ -37,14 +37,20 @@ export interface BuildOptions {
 	/**
 	 * Teardown the compiler WASM instance after build. This can improve performance when
 	 * building once, but may cause a performance hit if building multiple times in a row.
+	 *
+	 * @internal only used for testing
+	 * @default true
 	 */
 	teardownCompiler?: boolean;
 }
 
-/** `astro build` */
+/**
+ * Builds your site for deployment. By default, this will generate static files and place them in a dist/ directory.
+ * If SSR is enabled, this will generate the necessary server files to serve your site.
+ */
 export default async function build(
```

Comment:
Here's a thought that I am concerned about. 

It makes sense that `dev` and `preview` don't return anything, but I think `build` at least should return an array or a map of the assets that have been generated during the build, otherwise it wouldn't add enough value VS the CLI.

Reply:
> Maybe it would make sense to put these behind an experimental flag so we can change the returned signatures freely in the future.

Clever! BTW feel free to merge it; we can modify the return types, etc. in future PRs 😄